packages/dev/core/src/AudioV2/webAudio/components/webAudioParameterComponent.ts

```diff
@@ -105,6 +119,35 @@ export class _WebAudioParameterComponent {
         this._param.cancelScheduledValues(startTime);
         this._param.setValueCurveAtTime(_GetAudioParamCurveValues(shape, this._targetValue, (this._targetValue = value)), startTime, duration);
 
+        this._removeDeferredRamp();
+
         this._rampEndTime = startTime + duration;
     }
+
+    private _deferRamp(value: number, duration: number, shape: AudioParameterRampShape): void {
+        this._deferredRampOptions.duration = duration;
+        this._deferredRampOptions.shape = shape;
+        this._deferredTargetValue = value;
+
+        if (!this._deferredRampObserverable) {
+            if (this._engine.state !== "running") {
```

Comment:
I wonder if this could be simplified to always use the same observable as it might be harder in maintenance ?

Reply:
Good call! In the case where the audio engine is locked, the logic can be moved to the audio engine class, instead.